### ISSUE TITLE
Neural network backpropagation

### ISSUE THREAD

**Issue by midwestmonika** [cisc3410/4-review#16]
When do you use upward gradient and when do you multiply by -1 for the downward gradient?
